app/ide-desktop/eslint.config.js

```diff
@@ -137,14 +133,6 @@ const RESTRICTED_SYNTAXES = [
         selector: `TSAsExpression:not(:has(TSTypeReference > Identifier[name=const]))`,
         message: 'Avoid `as T`. Consider using a type annotation instead.',
     },
-    {
```

Comment:
This rule makes no sense with '[exactOptionalPropertyTypes](https://www.typescriptlang.org/tsconfig/exactOptionalPropertyTypes.html)`.

Reply:
`exactOptionalPropertyTypes` is not related to this rule. if anything, `exactOptionalPropertyTypes` *removes* `undefined` from the types of optional properties (unless explicitly declared as `| undefined`), meaning that this rule shouldn't cause any problem at all.

Reply:
In case I want to allow passing smth: undefined,  I need to specify it explicitly, but by specifying it explicitly, I'm making eslint angry.

Reply:
@MrFlashAccount but the point of `undefined` is that you *don't* need to pass it explicitly. if you need to omit something, simply don't specify the property at all

Reply:
Often we use obect destruction pattern and checking if a property is not undefiend before passing it is tedious.